### DISCUSSION TITLE
Lesson 7: Foundry Breaks if You change your function's names, (WHY ?) !!!!

### DISCUSSION THREAD

**Discussion by smhamidi** [Cyfrin/foundry-full-course-cu#361] [Q&A] [answered]
Hi everyone, Hope you are all doing great
I have this weird problem with naming my **_test_** functions in foundry, in lesson 7, patrick wrote a test name **_testFundFailsWithoutEnoughETH()_** to check if **_fund()_** function revert in case of not giving an enough value, and IT WORKS just like the videos.
I changed the name of the function to **_testFailWithoutEnoughETH()_** and now the function does NOT WORK.

Patrick's Code:
```solidity
    function testFundFailWithoutEnoughETH() public {
        vm.expectRevert(); // this like tells solidity that we expect the next line to revert
        // fundMe.fund(); // send the function without enough eth, we must send at least MINIMUM_USD/ETH
    }
```

Terminal output for Patrik's code: (As expected, The test fails)
```bash
[⠑] Compiling...
[⠢] Compiling 1 files with 0.8.20
[⠆] Solc 0.8.20 finished in 1.30s
Compiler run successful!

Running 1 test for test/FundMeTest.t.sol:FundMeTest
[FAIL. Reason: Call did not revert as expected] testFundFailWithoutEnoughETH() (gas: 3045)
Test result: FAILED. 0 passed; 1 failed; 0 skipped; finished in 2.98ms
Ran 1 test suites: 0 tests passed, 1 failed, 0 skipped (1 total tests)

Failing tests:
Encountered 1 failing test in test/FundMeTest.t.sol:FundMeTest
[FAIL. Reason: Call did not revert as expected] testFundFailWithoutEnoughETH() (gas: 3045)

Encountered a total of 1 failing tests, 0 tests succeeded
```
---

My Code:
```solidity
    function testFailWithoutEnoughETH() public {
        vm.expectRevert(); // this like tells solidity that we expect the next line to revert
        // fundMe.fund(); // send the function without enough eth, we must send at least MINIMUM_USD/ETH
    }
```

Terminal output for My Code: (MY Code Does Not Fail)
```bash
[⠑] Compiling...
[⠒] Compiling 1 files with 0.8.20
[⠢] Solc 0.8.20 finished in 1.26s
Compiler run successful!

Running 1 test for test/FundMeTest.t.sol:FundMeTest
[PASS] testFailWithoutEnoughETH() (gas: 3112)
Test result: ok. 1 passed; 0 failed; 0 skipped; finished in 1.24ms
Ran 1 test suites: 1 tests passed, 0 failed, 0 skipped (1 total tests)
```
---
My big concern is that i am suppose to test my contracts, and now i dont know even if my test are reliable or not, because they may act different with different names.

Can someone help me?

**Comment by Naman1729** [comment, marked as answer]
Hey @smhamidi ,
I would like to bring to your attention a potential issue in your code regarding the test prefixes. It appears that the prefix "testFail.." you're using is opposite to the conventional "test.." prefix. This inconsistency may lead to confusion and incorrect interpretation of test results.

Typically, the "test.." prefix is used to indicate successful or passing tests, while the "testFail.." prefix could imply tests that are expected to fail or produce an error.

for more information you can go to this Link :
https://book.getfoundry.sh/forge/writing-tests

**Reply by smhamidi** [reply]
You are absolutely right.  
that answered my question. thank you @Naman1729

**Comment by smhamidi** [comment]
testFail is another prefix known to foundry, I accidentally used it in my code, what are the odds :)   

- testFail: The inverse of the test prefix - if the function does not revert, the test fails.
